src/DependencyInjection/Compiler/AddExtensionsPass.php

```diff
@@ -30,11 +30,9 @@ public function process(ContainerBuilder $container): void
 
         $definition = $container->findDefinition('knp_menu.factory');
 
-        if (!method_exists($container->getParameterBag()->resolveValue($definition->getClass()), 'addExtension')) {
-            throw new InvalidConfigurationException(sprintf(
-                'To use factory extensions, the service of class "%s" registered as knp_menu.factory must implement the "addExtension" method',
-                $definition->getClass()
-            ));
+        if (!\method_exists($container->getParameterBag()->resolveValue($definition->getClass()), 'addExtension')) {
+            $msg = 'To use factory extensions, the service of class "%s" registered as knp_menu.factory must implement the "addExtension" method';
```

Comment:
I'd rather keep the pattern inline in the `sprintf` call, which IDEs validate as having the right number of placehilders.

Reply:
Problem is that line is really too long. We're way over recommended 80 columns